### PR TITLE
Added support to download JDBC Driver from configured URL. New testcase for JDBC Driver deployment on EAP server.

### PR DESCRIPTION
**Pull request by hhovsepy**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit cfe045e8983fb5be16026f6c5cb6f874270bfdb8:

`cfme/tests/middleware/test_middleware.py`:
- :red_circle: Line 3:1: **F401** *'quit' imported but unused*
- :red_circle: Line 8:1: **F401** *'Quadicon' imported but unused*
- :red_circle: Line 8:1: **F401** *'Region' imported but unused*
- :red_circle: Line 8:1: **F401** *'list_acc' imported but unused*
- :red_circle: Line 9:1: **F401** *'paginator' imported but unused*
- :red_circle: Line 9:1: **F811** *redefinition of unused 'paginator' from line 8*
- :red_circle: Line 12:1: **F401** *'deferred_verpick' imported but unused*
- :red_circle: Line 12:1: **F401** *'testgen' imported but unused*
- :bangbang: Line 20:94: **E202** *whitespace before '}'*
- :heavy_exclamation_mark: Line 31:1: **W293** *blank line contains whitespace*
- :bangbang: Line 34:1: **E302** *expected 2 blank lines, found 1*
- :red_circle: Line 35:5: **N802** *function name should be lowercase*
- :heavy_exclamation_mark: Line 38:1: **W293** *blank line contains whitespace*
- :red_circle: Line 39:9: **N802** *function name should be lowercase*
- :heavy_exclamation_mark: Line 43:11: **W291** *trailing whitespace*
- :bangbang: Line 48:1: **E302** *expected 2 blank lines, found 1*
- :heavy_exclamation_mark: Line 72:92: **W291** *trailing whitespace*
- :bangbang: Line 82:101: **E501** *line too long (101 > 100 characters)*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8946de874f2a0a553ee3c8db4397cce60c13d2e7:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/middleware/__init__.py [hunk 1]
```diff
@@ -117,6 +131,52 @@ def add_deployment(self, filename, runtime_name=None, enable_deploy=True, cancel
...
+            filename: Full path to JDBC Driver to import.
+            driver_name: Name of newly created JDBC Driver.
+            module_name: Name on Module to register on server side.
+            driver_class: JDBC Driver Class.
+            major_version: Major version of JDBC driver, optional.
+            minor_version: Minor version of JDBC driver, optional.
+            cancel: Whether to click Cancel instead of commit.
+        """
+        self.load_details(refresh=True)
+        jdbc_btn("Add JDBC Driver")
```
Comment (psav) [inline cfme/middleware/__init__.py]:
```python
        fill(jdbc_driver_form,
            {
                "file_select": filename,
                "jdbc_driver_name": driver_name,
                "jdbc_module_name": module_name,
                "jdbc_driver_class": driver_class,
                "major_version": major_version,
                "minor_version": minor_version
            })
```
Fill will do nothing if the value is None

File: cfme/tests/middleware/jdbc_driver_methods.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+
+def deploy_jdbc_driver(provider, server, file_path, driver_name, module_name,
+                       driver_class, major_version=None, minor_version=None):
+    """Deploys JDBC driver file into provided server.
+    Refreshes the provider relationships via UI operation.
+    """
+    server.add_jdbc_driver(file_path, driver_name, module_name, driver_class,
+                           major_version, minor_version)
+    provider.refresh_provider_relationships(method='ui')
```
Comment (psav) [inline cfme/tests/middleware/jdbc_driver_methods.py]: This could be done via REST to speed things up a little?

**Comment by psav** [comment]
Didn't mean to hit Approve sorry.

**Comment by dajoRH** [comment]
Lint report for commit c907e98ea8ad14aa8a8f2755158c38fc7da915ff:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Pull request merged**